tests/foreman/cli/test_realm.py

```diff
@@ -0,0 +1,327 @@
+# -*- encoding: utf-8 -*-
+# pylint: disable=no-self-use
+"""Test class for Realm CLI
+
+:Requirement: Realm
+
+:CaseAutomation: Automated
+
+:CaseLevel: Acceptance
+
+:CaseComponent: CLI
+
+:TestType: Functional
+
+:CaseImportance: High
+
+:Upstream: No
+"""
+import random
+
+from fauxfactory import gen_string
+from robottelo.cleanup import capsule_cleanup
+from robottelo.cli.base import CLIReturnCodeError
+from robottelo.cli.factory import (
+    make_location,
+    make_proxy,
+    make_realm,
+    CLIFactoryError,
+)
+from robottelo.cli.realm import Realm
+from robottelo.decorators import (
+    tier1,
+)
+from robottelo.test import CLITestCase
+
+
+class RealmTestCase(CLITestCase):
+    """Tests for Realms via Hammer CLI, must be run on QE RHEL7 Sat6.3 host"""
+
+    def _make_proxy(self, options=None):
+        """Create a Proxy and register the cleanup function"""
+        proxy = make_proxy(options=options)
+        # Add capsule to cleanup list
+        self.addCleanup(capsule_cleanup, proxy['id'])
+        return proxy
+
+    @tier1
+    def test_positive_create_delete_name(self):
```

Comment:
In case Realm is absolutely unique entity, then you can move `realm_name = gen_string('alpha', random.randint(1, 30))` into `SetUpClass` section and then add cleanup by name